pkg/iotools/hashReaderWriter.go

```diff
@@ -0,0 +1,152 @@
+package iotools
```

Comment:
What is the purpose of all of this? 🤔

Reply:
decorator pattern - with this, we can calculate during read/write operation multiple different hash values at once without the need to re-read per hash.

In most mvn-repos there are still MD5 and sha1 hashes used, but rest of OCM normally uses sha256... so I wanted to have all of them at once in place